oas_docs/README.md

```diff
@@ -1,9 +1,164 @@
+
+## Bundling commands
+
+Scripts in the `scripts` folder bundle individual Kibana OpenAPI bundles. There is a makefile to simplify the workflow. The following makefile targets are available
+
+| Command                    | Description                                                                                                                                                                                                        |
+| -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `api-docs`                 | Builds ESS Kibana OpenAPI bundle                                                                                                                                                                                   |
+| `api-docs-serverless`      | Builds Serverless Kibana OpenAPI bundle                                                                                                                                                                            |
+| `api-docs-lint`            | Lints built result bundles                                                                                                                                                                                         |
+| `api-docs-lint-errs`       | Lints built result bundles for errors                                                                                                                                                                              |
+| `api-docs-preview`         | Generates (ESS + Serverless) Kibana OpenAPI bundles preview                                                                                                                                                        |
+| `api-docs-overlay`         | Applies [overlays](https://docs.bump.sh/help/specification-support/overlays/) from `overlays` folder to the Kibana OpenAPI bundles and generate `*.new.yaml` files. Overlays help to fine tune the result bundles. |
+| `api-docs-overlay-preview` | Generates a preview for bundles produced by `api-docs-overlay`                                                                                                                                                     |
+
+## Bundling automation
+
+To have always up to date Kibana OpenAPI bundles bundling runs as a part of `Checks` step in CI on every PR build and on merge builds. If there are any changes in source OpenAPI files these changes get to the updated Kibana OpenAPI bundles and CI commits the changes. In that case CI is marked as failed and needs to be restarted.
```

Comment:
```suggestion
To keep the final resulting bundles always up-to-date and in sync with the source OpenAPI specs, the bundling runs as part of the `Checks` step in CI on every PR build and on merge builds. If there are any changes to the source OpenAPI files, these changes get propagated to the final bundles and CI commits the changes. In that case the build is marked as failed and needs to be restarted.
```

Do we really push commits from CI in this case, or just fail without committing?